File: ChipmeoApis/ChipmeoApis.Core/Entities/Combo.cs
namespace ChipmeoApis.Core.Entities;

public partial class Combo
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public decimal ComboPrice { get; set; }

    public string? Description { get; set; }

    public string? ImageUrl { get; set; }

    public bool? IsActive { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual ICollection<ComboItem> ComboItems { get; set; } = new List<ComboItem>();

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: foodstore-store/src/lib/types/combo.ts
export interface Combo {
	id: number;
	name: string;
	description?: string;
	imageUrl?: string | null;
	comboPrice: number;
	items: ComboItem[];
	isActive: boolean;
	createdAt?: string;
	updatedAt?: string;
}

export interface ComboItem {
	menuItemId: number;
	quantity: number;
}

export interface ComboCreateDto {
	name: string;
	description?: string;
	imageUrl?: string | null;
	comboPrice: number;
	items: ComboItem[];
	isActive: boolean;
}

export interface ComboUpdateDto {
	name: string;
	description?: string;
	imageUrl?: string | null;
	comboPrice: number;
	items: ComboItem[];
	isActive: boolean;
}
